norfair/drawing/draw_boxes.py

```diff
@@ -0,0 +1,178 @@
+
+    Parameters
+    ----------
+    frame : np.ndarray
+        The OpenCV frame to draw on. Modified in place.
+    drawables : Sequence[Union[Detection, TrackedObject]]
+        List of objects to draw, Detections and TrackedObjects are accepted.
+        This objects are assumed to contain 2 bi-dimensional points defining
+        the bounding box as `[[x0, y0], [x1, y1]]`.
+    color : ColorLike, optional
+        This parameter can take:
+        1. A color as a tuple of ints describing the BGR `(0, 0, 255)`
+        2. A 6-digit hex string `"#FF0000"`
+        3. One of the defined color names `"red"`
+        4. A string defining the strategy to choose colors from the Palette:
+            1. based on the id of the objects `"by_id"`
+            2. based on the label of the objects `"by_label"`
+        Note that if your objects don't have labels or ids (Detections never have ids)
+        the selected color will be the same for all objects.
+    thickness : Optional[int], optional
+        Thickness or width of the line.
+    draw_labels : bool, optional
+        If set to True, the label is added to a title that is drawn on top of the box.
+        If an object doesn't have a label this parameter is ignored.
+    draw_ids : bool, optional
+        If set to True, the id is added to a title that is drawn on top of the box.
+        If an object doesn't have an id this parameter is ignored.
+    text_size : Optional[float], optional
+        Size of the title, the value is used as a multiplier of the base size of the font.
+        By default the size is scaled automatically based on the frame size.
+    text_color : Optional[int], optional
+        Color of the text. By default the same color as the box is used.
+    text_thickness : Optional[int], optional
+        Thickness of the font. By default it's scaled with the `text_size`.
+
+    Returns
+    -------
+    np.ndarray
+        The resulting frame.
+    """
+    #
+    # handle deprecated parameters
+    #
+    if random_color is not None:
+        warn_once(
+            'Parameter "random_color" is deprecated, set `color="random"` instead'
+        )
+        color = "random"
+    if color_by_label is not None:
+        warn_once(
+            'Parameter "color_by_label" is deprecated, set `color="by_label"` instead'
+        )
+        color = "by_label"
+    if detections is not None:
+        warn_once('Parameter "detections" is deprecated, use "drawables" instead')
+        drawables = detections
+    if line_color is not None:
+        warn_once('Parameter "line_color" is deprecated, use "color" instead')
+        color = line_color
+    if line_width is not None:
+        warn_once('Parameter "line_width" is deprecated, use "thickness" instead')
+        thickness = line_width
+    if label_size is not None:
+        warn_once('Parameter "label_size" is deprecated, use "text_size" instead')
+        text_size = label_size
+    # end
+
+    if color is None:
```

Comment:
`color` shouldn't be `None` right? At least the type hint doesn't mention it.

Reply:
It shouldn't but it can happen when this function is called in  `draw_tracked_boxes` so I added this here to make the function a bit more robust